### PR TITLE
Feature append

### PR DESCRIPTION
**Pull request by jhannes**
I added the append command to save messages to a mailbox, following the coding style and documentation style of the rest of the codebase.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -172,6 +200,10 @@ message(s) in the currently open mailbox to another mailbox. `source` correspond
 specifies the messages to be moved. When completed, either calls the provided callback with signature `(err)`, or
 resolves the returned promise.
 
+- **append**(<*mixed*> message, <*object*> options, [<*function*> callback]) - *Promise* - Appends the argument
+message to the currently open mailbox to another mailbox. `message` is a RFC-822 compatible MIME message. Valid `options` are *mailbox*, *flags* and *date*. When completed, either calls the provided callback with signature `(err)`, or
```
Comment (chadxz) [inline README.md]: currently open mailbox **or** another mailbox

File: lib/imapSimple.js [hunk 2]
```diff
@@ -377,6 +377,35 @@ ImapSimple.prototype.delFlags = function (uid, flags, callback) {
...
 
 /**
+ * Appends a mime-encoded message to a mailbox
+ *
+ * @param {string|Array} message The messages uid
+ * @param {string|Array} options Option "mailbox" specifies mailbox, option "flag" indicate custom or predefined flag,
+     such as '\\Draft', '\\Unread'
+ * @param {function} [callback] Optional callback, receiving signature (err)
+ * @returns {undefined|Promise} Returns a promise when no callback is specified, resolving when the action succeeds.
+ * @memberof ImapSimple
```
Comment (chadxz) [inline lib/imapSimple.js]:
```
@param {string} message The message to append to the mailbox
@param {object} [options]
@param {string} [options.mailbox] The mailbox to append the message to. 
  Defaults to the currently open mailbox.
@param {string|Array<string>} [options.flags] A single flag (e.g. 'Seen') or an array 
  of flags (e.g. ['Seen', 'Flagged']) to append to the message. Defaults to
  no flags.
```

The rest of the docs look good

File: README.md [hunk 3]
```diff
@@ -172,6 +200,10 @@ message(s) in the currently open mailbox to another mailbox. `source` correspond
 specifies the messages to be moved. When completed, either calls the provided callback with signature `(err)`, or
 resolves the returned promise.
 
+- **append**(<*mixed*> message, <*object*> options, [<*function*> callback]) - *Promise* - Appends the argument
```
Comment (chadxz) [inline README.md]: please mark the options object as optional too

**Review by chadxz — CHANGES_REQUESTED** [review]
Thank you for the pull request! I think this is a good addition to the library, and appreciate you taking the time to add it. I've got a few requests but overall it looks good.

**Comment by jhannes** [comment]
I've made the requested changes. Thanks for teaching me about `@param ... options.flags`. I didn't realize that I could document like this and it's much better! :-)

**Review by chadxz — APPROVED** [review]

**Comment by chadxz** [comment]
Released as imap-simple@3.2.0

**Comment by chadxz** [comment]
Thank you for your contribution 🍻

**Comment by jhannes** [comment]
🍻

**Pull request merged**